pulser-core/pulser/waveforms.py

```diff
@@ -804,6 +806,18 @@ class InterpolatedWaveform(Waveform):
             interpolator class.
     """
 
+    def __new__(cls, *args, **kwargs):  # type: ignore
```

Comment:
nit: I think that you can replicate the arguments in `__init__()` here, that way you don't have to do the tricks with args and kwargs below. Something like this should work:
```suggestion
    def __new__(
        cls,
        duration: Union[int, Parametrized],
        values: Union[ArrayLike, Parametrized],
        times: Optional[Union[ArrayLike, Parametrized]],
        *args,
        **kwargs,
    ):
```
With `__new__()` I'm never sure without testing though
If this is too much trouble, just leave it be